src/lib/WalletInfoCollector.ts

```diff
@@ -92,14 +92,19 @@ export default class WalletInfoCollector {
             derivedAccountsPromise = WalletInfoCollector._deriveAccounts(startIndex,
                 ACCOUNT_MAX_ALLOWED_ADDRESS_GAP, walletType, walletId);
 
-            // find accounts with a balance > 0
-            // TODO should use transaction receipts
+            // find accounts with a balance > 0 and/or transaction receipts
             foundAccounts = [];
             const balances = await WalletInfoCollector._getBalances(derivedAccounts!);
             for (const account of derivedAccounts!) {
                 const balance = balances.get(account.address);
-                if (balance === undefined || balance === 0) continue;
-                foundAccounts.push(account);
+                if (balance !== undefined && balance !== 0) {
+                    foundAccounts.push(account);
+                    continue;
+                }
+                const receipts = await NetworkClient.Instance.requestTransactionReceipts(account.address);
```

Comment:
Instead of `await` in the for loop, use `Promise.all` by executing the loop code as promises for each derived account